lib/pure/os.nim

```diff
@@ -1648,6 +1648,31 @@ proc setFilePermissions*(filename: string, permissions: set[FilePermission],
       var res2 = setFileAttributesA(filename, res)
     if res2 == - 1'i32: raiseOSError(osLastError(), $(filename, permissions))
 
+proc isAdmin*: bool {.noWeirdTarget.} =
+  ## Returns whether the caller's process is a member of the Administrators local
+  ## group (on Windows) or a root (on POSIX), via `geteuid() == 0`.
+  when defined(windows):
+    # Rewrite of the example from Microsoft Docs:
+    # https://docs.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-checktokenmembership#examples
+    var b: WINBOOL
+    var ntAuthority = SID_IDENTIFIER_AUTHORITY(value: [
+      BYTE(0), BYTE(0), BYTE(0), BYTE(0), BYTE(0), BYTE(SECURITY_NT_AUTHORITY)
+    ])
+    var administratorsGroup: PSID
+    b = AllocateAndInitializeSid(addr ntAuthority,
+                                 BYTE(2),
+                                 SECURITY_BUILTIN_DOMAIN_RID,
+                                 DOMAIN_ALIAS_RID_ADMINS,
+                                 0, 0, 0, 0, 0, 0,
+                                 addr administratorsGroup)
+    if bool(b):
```

Comment:
instead of all those `bool(b)`, about introducing a `isSuccess` which is more self-documenting and more typesafe (in future we may make `WINBOOL` a distinct int32 as it's error-prone to have different convention (0 vs non-zero for successs across os)

```nim
# in winlean:
proc isSuccess*(a: WINBOOL): bool {.inline.} = bool(a)
```

Reply:
I agree. Fixed.